challenges/lib/src/main/java/challenges/utilities/AnimalShelter.java

```diff
@@ -0,0 +1,40 @@
+package challenges.utilities;
+
+import challenges.stacksandqueues.Queue;
+
+import java.util.Locale;
+
+public class AnimalShelter<T> {
+    Queue dogs = new Queue<T>();
+    Queue cats = new Queue<T>();
+
+    public void enqueue(T animal) throws ClassNotFoundException {
```

Comment:
Good work on making it accept a generic animal, instead of writing out two separate methods!